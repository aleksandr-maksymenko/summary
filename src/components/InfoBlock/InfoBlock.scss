@import 'src/styles/mixins';
@import 'src/styles/variables';

.info-block {
  padding: {
    top: 1rem;
    bottom: 1rem;
  }
  background-color: #e9e9e9;

  &__wrapper {
    max-width: $container;
    margin: 0 auto;
  }

  &__contact {
    width: 100%;
    margin-bottom: 1.5rem;
    display: flex;
    align-items: center;

    &:last-of-type {
      margin-bottom: 0;
    }

    @include mobile {
      width: 50%;

      &:last-of-type,
      &:nth-last-of-type(2) {
        margin-bottom: 0;
      }
    }

    &-list {
      padding: 0 1.25rem;
      display: flex;
      flex-direction: column;
      flex-wrap: wrap;

      @include mobile {
        flex-direction: row;
      }
    }
  }

  &__link {
    font-size: 1rem;
    text-decoration: none;
    color: #1d1d1d;

    &:visited {
      color: #969696;
    }
  }

  .svg__icon {
    max-width: 1.5rem;
    max-height: 1.5rem;
    margin-right: 0.5rem;
  }
}
