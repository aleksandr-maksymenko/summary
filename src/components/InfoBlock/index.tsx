import React from 'react'

import Linkedin from '@icons/Linkedin'
import Mail from '@icons/Mail'
import Location from '@icons/Location'
import Phone from '@icons/Phone'
import Skype from '@icons/Skype'
import Telegram from '@icons/Telegram'
import Viber from '@icons/Viber'
import Git from '@icons/Git'

const contacts = [
  {
    type: 'email',
    link: 'mailto:oleksandr.maksymenko@outlook.com',
    icon: <Mail />,
    text: 'oleksandr.maksymenko@outlook.com',
  },
  {
    type: 'phone',
    link: 'tel:+380633167816',
    icon: <Phone />,
    text: '+380 (063) 316-78-16',
  },
  {
    type: 'telegram',
    link: 'https://telegram.me/oleksandr_maksymenko',
    icon: <Telegram />,
    text: 'Message me into telegram',
  },
  {
    type: 'viber',
    link: 'viber:+380633167816',
    icon: <Viber />,
    text: 'Message me into viber',
  },
  {
    type: 'skype',
    link: 'https://join.skype.com/invite/mm0nz26gOh3S',
    icon: <Skype />,
    text: 'Oleksandr Maksymenko',
  },
  {
    type: 'location',
    link: 'https://www.google.com.ua/maps/@50.4273936,30.5173127,14z',
    icon: <Location />,
    text: 'Kiev, Ukraine',
  },
  {
    type: 'linkedin',
    link: 'https://www.linkedin.com/in/oleksandr-maksymenko',
    icon: <Linkedin />,
    text: 'oleksandr-maksymenko',
  },
  {
    type: 'github',
    link: 'https://github.com/oleksandrmaksymenko',
    icon: <Git />,
    text: 'GitHub',
  },
]

export default function InfoBlock() {
  const defaultInfoBlockClassName = 'info-block'

  return (
    <div className={defaultInfoBlockClassName}>
      <div className={`${defaultInfoBlockClassName}__wrapper`}>
        <ul className={`${defaultInfoBlockClassName}__contact-list`}>
          {contacts.map(contact => (
            <li
              key={contact.type}
              className={`${defaultInfoBlockClassName}__contact`}
            >
              {contact.icon}
              <a
                href={contact.link}
                className={`${defaultInfoBlockClassName}__link`}
              >
                {contact.text}
              </a>
            </li>
          ))}
        </ul>
      </div>
    </div>
  )
}
