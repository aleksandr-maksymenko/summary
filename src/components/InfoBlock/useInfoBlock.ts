import { useSelector } from 'react-redux'
import { RootState } from 'src/store'
import { ContactType } from 'src/types'

const useInfoBlock = () => {
  const contacts = useSelector(
    (state: RootState) => state.contacts,
  ) as ContactType[]

  return { contacts }
}

export default useInfoBlock
