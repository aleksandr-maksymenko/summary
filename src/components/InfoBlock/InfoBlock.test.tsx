import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import InfoBlock from './index'

describe('About', () => {
  let wrapper: ShallowWrapper

  beforeAll(() => {
    wrapper = shallow(<InfoBlock />)
  })

  it('Should return section', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
