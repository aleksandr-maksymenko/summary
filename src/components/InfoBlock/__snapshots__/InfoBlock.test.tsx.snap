// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Should return section 1`] = `ShallowWrapper {}`;
